contracts/base/LoansBase.sol

```diff
@@ -241,6 +247,10 @@ contract LoansBase is LoansInterface, Base {
 
         borrowerLoans[request.borrower].push(loanID);
 
+        if (request.userId != bytes32(0)) {
+            userIdHasLoan[request.userId] = true;
+        }
```

Comment:
We'll need to validate that this flag is not already set OR if it is check their previous unsecured loan status. If its `TermsSet`, meaning it was never taken out, validate that the terms have expired.

Also I do not see a check on the `collateralRatio` value @tpscrpt this should only happen when the collateral ratio is 0, right?